src/Analyzers/CSharp/Analyzers/UseCompoundAssignment/CSharpUseCompoundAssignmentDiagnosticAnalyzer.cs

```diff
@@ -24,5 +25,34 @@ protected override SyntaxKind GetAnalysisKind()
         protected override bool IsSupported(SyntaxKind assignmentKind, ParseOptions options)
             => assignmentKind != SyntaxKind.CoalesceExpression ||
             ((CSharpParseOptions)options).LanguageVersion >= LanguageVersion.CSharp8;
+
+        protected override int TryGetIncrementOrDecrement(SyntaxKind opKind, object constantValue)
+        {
+            if (constantValue is string || constantValue is not IConvertible convertible)
+            {
+                return 0;
+            }
+
+            if (convertible.ToDouble(null) == 1)
```

Comment:
Can ToDouble throw?

Reply:
I'm not sure what's the best way to test for all "numeric 1"s.
Since the list is exhaustive and not so long, maybe `is (byte)1 or (short)1 or (int)1 or ...` is acceptable?

Reply:
1. ToDouble might be fine.  But if it can throw, that has to be accounted for :)
2. Making an exhaustive list is also fine.  We have an IntegerUtilities type that can help.  Though i think you'll still need to do the FP checks yoruself.